### PR TITLE
Move cmath into `sys`

### PR DESCRIPTION
**Pull request by joboet**
Part of #117276.

r? @ChrisDenton

### CODE REVIEW

File: library/std/src/sys/cmath/mod.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#![cfg(not(test))]
+
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "windows")] {
+        mod windows;
+        pub use windows::*;
```
Comment (ChrisDenton) [inline library/std/src/sys/cmath/mod.rs:6]: Should we keep platform specific things in pal?

Comment (joboet) [reply library/std/src/sys/cmath/mod.rs:6]: If they implement a shared interface, I'd say no, so that similar code stays together.

Comment (ChrisDenton) [reply library/std/src/sys/cmath/mod.rs:6]: Ok, that makes sense to me as a principle.

**Comment by ChrisDenton** [comment]
r=me when conflicts are fixed.

**Comment by joboet** [comment]
@ChrisDenton
All rebased.

**Pull request merged**